lib/internal/test_runner/runner.js

```diff
@@ -248,14 +220,79 @@ class FileTest extends Test {
       this.#buffer = [];
     }
   }
-  addToReport(ast) {
-    this.#accumulateReportItem(ast);
+  addToReport(item) {
+    this.#accumulateReportItem(item);
     if (!this.isClearToSend()) {
-      ArrayPrototypePush(this.#buffer, ast);
+      ArrayPrototypePush(this.#buffer, item);
       return;
     }
     this.#drainBuffer();
-    this.#handleReportItem(ast);
+    this.#handleReportItem(item);
+  }
+  parseMessage(readData) {
+    if (readData.length === 0) return;
+
+    ArrayPrototypePush(this.#messageBuffer, readData);
+    this.#messageBufferSize += readData.length;
```

Comment:
We should avoid calling the getter twice here.
```suggestion
    const dataLength = TypedArrayPrototypeGetLength(readData);
    if (dataLength === 0) return;

    ArrayPrototypePush(this.#messageBuffer, readData);
    this.#messageBufferSize += dataLength;
```